File: articles/the-history-of-data-structures-and-algorithm-analysis/go/growth_rates.go
package main

import (
	"fmt"
	"math"
)

func main() {
	for _, n := range []float64{10, 100, 1000, 10000} {
		fmt.Printf("n=%.0f, log2=%.6f, nlogn=%.6f, n2=%.0f\n", n, math.Log2(n), n*math.Log2(n), n*n)
	}
}

File: articles/the-history-of-data-structures-and-algorithm-analysis/cpp/growth_rates.cpp
#include <cmath>
#include <iostream>
#include <vector>

int main() {
    for (double n : std::vector<double>{10, 100, 1000, 10000}) {
        std::cout << "n=" << n << ", log2=" << std::log2(n) << ", nlogn=" << n * std::log2(n) << ", n2=" << n*n << "\n";
    }
}
